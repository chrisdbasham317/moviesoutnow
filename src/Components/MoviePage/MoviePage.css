.moviePage__div {
    display: flex;
    position: relative;
    z-index: -1;
    background: rgba(8, 8, 8, 0.6);
    justify-content: center;
}

.moviePage--backdrop {
    width: 100%;
    opacity: .2;
}

.movie--poster--details__div {
    display: flex;
    position: absolute;
    top: 0px;
    height: 100%;
    width: 80%;
    align-items: center;
}

.moviePage--poster {
    height: 85%;
}

.movie__details {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    /* align-content: flex-start; */
    align-items: flex-start;
    color: #FFF;
    height: 70%;
    margin-left: 25px;
}

.movie--title--year,
.movie--user--score--fav--votes {
    display: flex;
    justify-content: space-evenly;
}

.movie--title--year {
    align-items: center;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.movie--title {
    font-size: 40px;
    margin: 0;
}

.movie--year {
    font-size: 40px;
    color:	#BEBEBE;
    margin-left: 20px;
    margin: 0;
}

.movie--user--score--fav--votes {
    width: 80%;
    display: flex;
    justify-content: space-evenly;
    font-size: 30px;
}