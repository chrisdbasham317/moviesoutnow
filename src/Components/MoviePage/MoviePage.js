import React from 'react';
import './MoviePage.css'

const MoviePage = ({ title, poster_path, backdrop_path, release_date, vote_average, vote_count, overview, genre_ids, isFavorite, id }) => {
    getGenres = () => {
        reutrn "genre"
    }

    return (
        <div className="moviePage__div" key={id}>
            <img 
                src={`https://image.tmdb.org/t/p/w500${backdrop_path}`} 
                alt={`${title} Back Drop`}
                className="moviePage--backdrop" />
            <div className="movie--poster--details__div">
                <img
                src={`https://image.tmdb.org/t/p/w500${poster_path}`} 
                alt={`${title} Poster`}
                className="moviePage--poster" />
                <div className="movie__details">
                    <div className="movie--title--year">
                        <h3 className="movie--title">{title}</h3>
                        <p className="movie--year">{release_date.split('-')[0]}</p>
                    </div>
                    <div className="movie--user--score--fav--votes">
                        <p className="movie--score">User Score: {vote_average.split('.').join('')}</p>
                        <p className="movie--favorited">Fav</p>
                        <p className="movie--vote">Votes: {vote_count}</p>
                    </div>
                    <p className="movie--overview">{overview}</p>
                    <div className="movie--genres">{getGenres}</div>
                </div>
            </div>
        </div>
    ) 
}

export default MoviePage