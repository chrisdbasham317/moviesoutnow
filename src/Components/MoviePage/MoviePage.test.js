import React from 'react';
import { shallow } from 'enzyme';
import { MoviePage } from './MoviePage';

describe("MoviePage", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <MoviePage
        movies={[
          {
            popularity: 458.719,
            vote_count: 4305,
            video: false,
            poster_path: "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
            id: 475557,
            adult: false,
            backdrop_path: "/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg",
            original_language: "en",
            original_title: "Joker",
            genre_ids: [
              80,
              18,
              53
            ],
            title: "Joker",
            vote_average: 8.6,
            overview: "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
            "release_date": "2019-10-04"
          },
          {
            popularity: 231.895,
            vote_count: 472,
            video: false,
            poster_path: "/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg",
            id: 420809,
            adult: false,
            backdrop_path: "/skvI4rYFrKXS73BJxWGH54Omlvv.jpg",
            original_language: "en",
            original_title: "Maleficent: Mistress of Evil",
            genre_ids: [
              12,
              14,
              10751
            ],
            title: "Maleficent: Mistress of Evil",
            vote_average: 7.2,
            overview: "Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.",
            "release_date": "2019-10-18"
          }
        ]}
        genres={[{
          id: 12,
          name: 'Action'
        },
        {
          id: 14,
          name: 'Comedy'
        },
        {
          id: 10751,
          name: 'Horror'
        }]}
        currentUser={{ id: 3, name: 'Chris' }}
        currentMovie={420809}
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})