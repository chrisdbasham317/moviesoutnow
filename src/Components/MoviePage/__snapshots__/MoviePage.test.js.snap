// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should match the snapshot 1`] = `
<div
  className="moviePage__div"
  key="420809"
>
  <img
    alt="Maleficent: Mistress of Evil Back Drop"
    className="moviePage--backdrop"
    src="https://image.tmdb.org/t/p/w500/skvI4rYFrKXS73BJxWGH54Omlvv.jpg"
  />
  <div
    className="movie--poster--details__div"
  >
    <img
      alt="Maleficent: Mistress of Evil Poster"
      className="moviePage--poster"
      src="https://image.tmdb.org/t/p/w500/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg"
    />
    <div
      className="movie__details"
    >
      <div
        className="movie--title--year"
      >
        <p
          className="movie--year"
        >
          2019
        </p>
        <h3
          className="movie--title"
        >
          Maleficent: Mistress of Evil
        </h3>
      </div>
      <div
        className="movie--user--score--fav--votes"
      >
        <div
          className="movie--score__div"
        >
          <p
            className="movie--score--text"
          >
            Score:
          </p>
          <p
            className="movie--score"
          >
            72
            %
          </p>
        </div>
        <div
          className="movie--vote__div"
        >
          <p
            className="movie--vote--text"
          >
            Votes:
          </p>
          <p
            className="movie--vote"
          >
            472
          </p>
        </div>
      </div>
      <p
        className="movie--overview"
      >
        Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.
      </p>
      <div
        className="movie--genres"
      >
        <span
          className="genre-title"
        >
          Genres:
        </span>
          
        Action, Comedy, Horror
      </div>
    </div>
  </div>
</div>
`;
